### PR TITLE
Fixes for latest rust

### PR DESCRIPTION
**Pull request by andrebeat**
Fixes for rustc 1.0.0-nightly (b47aebe3f 2015-02-26) (built 2015-02-26).

### CODE REVIEW

File: tests/single.rs [hunk 1]
```diff
@@ -50,23 +54,16 @@ describe! thunk {
...
             let counter = Arc::new(Mutex::new(0));
             let counter_clone = counter.clone();
-            let result =  Thread::scoped(move || {
+            thread::scoped(move || {
                 let value = Dropper(counter_clone);
                 let t = Thunk::<()>::new(move || {
                     // Get a reference so value is captured.
                     let _x = &value;
-
-                    panic!("Muahahahah")
```
Comment (andrebeat) [inline tests/single.rs]: It's no longer possible to handle panics from scoped threads ([PR #22435](https://github.com/rust-lang/rust/pull/22435)). I'm not entirely sure if the panic was relevant to the test logic.

Comment (reem) [inline tests/single.rs]: Yes, the panic is actually quite important, since this is testing the behavior of a thunk under a panic.

Comment (andrebeat) [inline tests/single.rs]: Makes sense, although I don't know how to fix it.

**Comment by andrebeat** [comment]
I've updated to rust 1.0.0-nightly (12b846ab8 2015-03-09) (built 2015-03-09). It is now possible to join spawned threads and handle panics, so the original unit test behavior is restored.

**Comment by reem** [comment]
Excellent! Thank you very much.

**Pull request merged**